### PR TITLE
KFSPTS-26275 Add CNCR Request V4 workflow handling

### PR DESCRIPTION
**Pull request by cah292**
This is an updated version of the declined #1376 changes regarding the new Concur API workflow handling. As was discussed in the user story, the new version of this code will always approve incoming Travel Requests, even if their accounts fail validation.

In addition, this PR will add an extra approve-anyway warning to the validation messages if a Request fails account validation. The message is not being added to the workflow action itself, but it will be added to the Event Notification V2 report.

Please let me know if you have any questions or concerns about this updated approach.

### CODE REVIEW

**Review by jhulslander — COMMENTED** [review]
changes look good, I noticed one thing that could be adjusted

File: src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurRequestV4ServiceImpl.java [hunk 1]
```diff
@@ -224,10 +238,106 @@ protected void updateRequestStatusInConcur(
...
+                    + "The matching entries in the upcoming SAE file may get rejected as a result.");
+        }
+        
+        if (!shouldUpdateStatusInConcur()) {
+            LOG.info("updateRequestStatusInConcur, Concur workflow actions are currently disabled "
+                    + "in this KFS environment");
+            return;
+        }
+        
+        String requestId = resultsDTO.getReportNumber();
```
[the remark this answers is not in the document]
Comment (cah292) [reply src/main/java/edu/cornell/kfs/concur/batch/service/impl/ConcurRequestV4ServiceImpl.java:260]: Refactoring.

**Comment by cah292** [comment]
I've refactored the code for clarity as requested. I also ran the related unit tests locally (and adjusted them as needed) to ensure that things still work as expected.

This is ready for review again.

**Review by jhulslander — APPROVED** [review]
changes look good, unit tests pass, I shall merge

**Pull request merged**